### DISCUSSION TITLE
Make expression writing more natural.

### DISCUSSION THREAD

**Discussion by gzp79** [SeaQL/sea-query#165] [Ideas]
Have you considered implementing the +-* traits on simpleexpr and ident ? This way writing expressions might be more natural. Unfortunately == && || won't work.

Another option could be some macro: ` expression!( Persons.Age + 2 > value(12))`

**Comment by tyt2y3** [comment]
Yes. But there are plently of quirks, for example, [PartialEq::eq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) require that the return value is `bool`, which does not work for us.

A macro with a custom DSL is the only way to go. I myself am not an advocate of DSL, but I could accept a PR if someone came up with a brilliant design!
